File: app/src/main/java/com/lostincontext/ruledetails/items/FenceItemViewHolder.java
package com.lostincontext.ruledetails.items;


import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.lostincontext.commons.list.ViewHolder;
import com.lostincontext.databinding.ItemFenceBinding;

import java.util.List;

import static com.lostincontext.ruledetails.RuleDetailsContract.LINK_CHANGED;

public class FenceItemViewHolder extends ViewHolder {


    private ItemFenceBinding binding;

    public static FenceItemViewHolder create(LayoutInflater inflater,
                                             ViewGroup parent,
                                             FenceItemCallback callback) {
        ItemFenceBinding itemBinding = ItemFenceBinding.inflate(inflater,
                                                                parent,
                                                                false);
        return new FenceItemViewHolder(itemBinding, callback);
    }


    public FenceItemViewHolder(ItemFenceBinding binding, FenceItemCallback callback) {
        super(binding.getRoot());
        this.binding = binding;
        binding.setCallback(callback);
    }

    public void bindTo(FenceItem item, List<Object> payloads) {
        if (payloads.isEmpty()) binding.setItem(item);
        else {
            for (int i = 0, size = payloads.size(); i < size; i++) {
                if ((payloads.get(i)) == LINK_CHANGED) {
                    binding.linkTextView.setText(item.link.getResourceId());
                }

            }
        }
    }


}

File: app/src/main/java/com/lostincontext/ruledetails/items/FenceItemViewHolder.kt
package com.lostincontext.ruledetails.items


import android.view.LayoutInflater
import android.view.ViewGroup

import com.lostincontext.commons.list.ViewHolder
import com.lostincontext.databinding.ItemFenceBinding

import com.lostincontext.ruledetails.RuleDetailsContract.LINK_CHANGED

class FenceItemViewHolder(private val binding: ItemFenceBinding, callback: FenceItemCallback)
: ViewHolder(binding.root) {


    init {
        binding.callback = callback
    }

    fun bindTo(item: FenceItem, payloads: List<Any>) {
        if (payloads.isEmpty())
            binding.item = item
        else {
            var i = 0
            val size = payloads.size
            while (i < size) {
                if (payloads[i] === LINK_CHANGED) {
                    binding.linkTextView.setText(item.link.resourceId)
                }
                i++

            }
        }
    }

    companion object {

        fun create(inflater: LayoutInflater,
                   parent: ViewGroup,
                   callback: FenceItemCallback): FenceItemViewHolder {
            val itemBinding = ItemFenceBinding.inflate(inflater,
                                                       parent,
                                                       false)
            return FenceItemViewHolder(itemBinding, callback)
        }
    }


}
